### ISSUE TITLE
Truncated text when used inside markdown due to line-height

### ISSUE THREAD

**Issue by a-h** [zjffun/reveal.js-mermaid-plugin#5]
When using Markdown text:

````
### Target architecture

```mermaid
%%{init: {'theme': 'dark', 'themeVariables': { 'darkMode': true }}}%%
flowchart TD
  A[Start] --> B{Is it?};
  B -- Yes --> C[OK];
  C --> D[Rethink];
  D --> B;
  B -- No ----> E[End];
```
````

The line height is not correct, resulting in truncated text.

![image](https://user-images.githubusercontent.com/1029947/220128888-ad844ce4-540c-4e5f-9206-53dda9e9b344.png)

Removing the `line-height` CSS setting solves the problem:

![image](https://user-images.githubusercontent.com/1029947/220130280-41b34ae5-1466-4511-b897-3a5a2eebe7a4.png)

It seems that in this case, the Mermaid content inherits the preformatted config. Perhaps this plugin should do something to override the background CSS?

**Comment by a-h** [comment]
I think this post describes the issue, in that the plugin needs to replace the outer HTML rather than inner HTML? https://css-tricks.com/making-mermaid-diagrams-in-markdown/

**Comment by zjffun** [comment]
> I think this post describes the issue, in that the plugin needs to replace the outer HTML rather than inner HTML? https://css-tricks.com/making-mermaid-diagrams-in-markdown/

You are right, the style of the `pre` tag affects the display of mermaid. We can also set the style more precisely:

```html
<style type="text/css">
pre.mermaid {
  line-height: 1;
}
</style>
```

or direct write HTML in Markdown:

```
### Target architecture

<div class="mermaid">
  <pre>
    %%{init: {'theme': 'dark', 'themeVariables': { 'darkMode': true }}}%%
    flowchart TD
      A[Start] --> B{Is it?};
      B -- Yes --> C[OK];
      C --> D[Rethink];
      D --> B;
      B -- No ----> E[End];
  </pre>
</div>
```

As a plugin, it is difficult to determine how much scope and how high priority to the style to add. I think it is better to leave this matter to mermaid or users. Thank you.

**Comment by TimPietrusky** [comment]
@zjffun from my point of view, the plugin should already work in a way, that it works out of the box and provide the styling, so that the user doesn't need to change anything manual. 

I'm using mermaid like this:

```
<div class="mermaid">
  <pre>
    %%{init: {'theme': 'dark', 'themeVariables': { 'darkMode': true }}}%%
    flowchart TD
      A[Start] --> B{Is it?};
      B -- Yes --> C[OK];
      C --> D[Rethink];
      D --> B;
      B -- No ----> E[End];
  </pre>
</div>
```

And I had to add this style in order to not get truncated text:

```
<style type="text/css">
.mermaid .nodeLabel {
    line-height: 1;
    vertical-align: super;
}
</style>
```

**Comment by zjffun** [comment]
@TimPietrusky The default styles should be correct, it looks like we need to check if there are other libraries that set the style of `.nodeLabel`. LIke:

<img width="1277" alt="Screen Shot 2023-08-13 at 13 00 42" src="https://github.com/zjffun/reveal.js-mermaid-plugin/assets/25266120/3bee8687-3af1-4e47-84d3-0526a5ce72ce">

If a widely used library accidentally sets the style of `.nodeLabel`, the best way is to let the library which sets the style of `.nodeLabel` to fix, but this is usually difficult, so there is a high probability that this library also set the style of `.nodeLabel` to fix it.

**Comment by l0b0** [comment]
After looking into this with Firefox's dev tools, it looks like `height="16"` in `<foreignObject height="16" width="122.75"><div xmlns="http://www.w3.org/1999/xhtml" style="display: inline-block; white-space: nowrap;"><span class="nodeLabel">[my label]</span></div></foreignObject>` is the problem. Increasing that value was the thing which made labels display properly.
